docs/vectorizer/document-vectorization.md

```diff
@@ -0,0 +1,525 @@
+# Document Vectorization in PGAI
+
+## Introduction
+
+Document vectorization is a critical capability for AI applications that need to work with unstructured data like PDFs, Word documents, or other file formats stored outside PostgreSQL. While RAG (Retrieval Augmented Generation) applications typically require text data, real-world enterprise scenarios often involve documents that:
+
+- Are stored in external systems like S3 or local filesystems
+- Come in various formats (PDF, DOCX, XLSX, EPUB, etc.)
+- Contain complex structures like tables, images, and formatting
+- Change frequently, requiring synchronization between sources and embeddings
+
+PGAI's document vectorization system addresses these challenges through a declarative approach that handles loading, parsing, chunking, and embedding documents with minimal configuration. This architecture:
+
+- Keeps document metadata in PostgreSQL while document content lives in optimized storage systems
+- Automatically processes documents into LLM-friendly formats
+- Maintains synchronization between documents and their embeddings
+- Provides resilient error handling and monitoring
+
+## Setting Up Document Storage
```

Comment:
Nit: Maybe it is a personal preference, but I really hate "title case". It feels so unnatural to me 🤷